src/components/vm/disks/diskAdd.jsx

```diff
@@ -479,70 +463,70 @@ export class AddDiskModalBody extends React.Component {
+            setDiskParams(diskParams => ({ ...diskParams, storagePool: currentPool }));
             // Reset the format only when the Format selection dropdown changes entries - otherwise just keep the old selection
             // All pool types apart from 'disk' have either 'raw' or 'qcow2' format
             if (currentPool && prevPool && ((currentPool.type == 'disk' && prevPool.type != 'disk') || (currentPool.type != 'disk' && prevPool.type == 'disk'))) {
                 // use onValueChange instead of setState in order to perform subsequent state change logic
-                this.onValueChanged('format', getDefaultVolumeFormat(value));
+                onValueChanged('format', getDefaultVolumeFormat(value));
             }
 
-            if (this.state.mode === USE_EXISTING) { // user changed pool
+            if (mode === USE_EXISTING) { // user changed pool
                 // use onValueChange instead of setState in order to perform subsequent state change logic
-                this.onValueChanged('existingVolumeName', getDefaultVolumeName(value, storagePools, vm));
+                onValueChanged('existingVolumeName', getDefaultVolumeName(value, storagePools, vm));
             }
             break;
         }
         case 'existingVolumeName': {
-            stateDelta.existingVolumeName = value;
-            this.setState(prevState => { // to prevent asynchronous for recursive call with existingVolumeName as a key
-                return this.existingVolumeNameDelta(value, prevState.storagePoolName);
-            });
+            // to prevent asynchronous for recursive call with existingVolumeName as a key
+            setDiskParams(diskParams => ({ ...diskParams, ...existingVolumeNameDelta(value, storagePoolName) }));
             break;
         }
         case 'mode': {
-            this.setState(prevState => { // to prevent asynchronous for recursive call with existingVolumeName as a key
-                stateDelta = this.initialState;
-                stateDelta.mode = value;
-                if (value === USE_EXISTING) { // user moved to USE_EXISTING subtab
-                    const poolName = stateDelta.storagePoolName;
-                    if (poolName)
-                        stateDelta = { ...stateDelta, ...this.existingVolumeNameDelta(getDefaultVolumeName(poolName, storagePools, vm), prevState.storagePoolName) };
+            if (value === USE_EXISTING) { // user moved to USE_EXISTING subtab
+                const poolName = diskParams.storagePoolName;
+                if (poolName) {
+                    setMode(
+                        value,
+                        () => setDiskParams(diskParams => ({
+                            ...diskParams,
+                            ...initialDiskParams,
+                            ...existingVolumeNameDelta(getDefaultVolumeName(poolName, storagePools, vm), storagePoolName)
+                        })
+                        ));
+                    return;
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1303-20231109-142627-1ac1b404-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)